qiita_db/metadata_template/prep_template.py

```diff
@@ -507,11 +506,16 @@ def generate_files(self, samples=None, columns=None):
         with qdb.sql_connection.TRN:
             # figuring out the filepath of the prep template
             _id, fp = qdb.util.get_mountpoint('templates')[0]
+            # update timestamp in the DB first
+            qdb.sql_connection.TRN.add(
+                """UPDATE qiita.prep_template
+                   SET modification_timestamp = CURRENT_TIMESTAMP
+                   WHERE prep_template_id = %s""", [self._id])
+            ctime = self.modification_timestamp
             fp = join(fp, '%d_prep_%d_%s.txt' % (self.study_id, self._id,
-                      strftime("%Y%m%d-%H%M%S")))
+                      ctime.strftime("%Y%m%d-%H%M%S")))
```

Comment:
Should we query the `modification_timestamp` value so there's always a 1:1 match between the filenames and the database entries? Otherwise it seems like those two might be out of sync.

Reply:
That's right and that's what is doing AFAIK. We first insert the current time in the DB, then we retrieve the modification_timestamp (`ctime = self.modification_timestamp`) and we use ctime as the timestamp for the filename.

Reply:
Apologies, reading ctime made me think of the `ctime` function - my bad!